jsonrpc2_test.go

```diff
@@ -314,105 +314,88 @@ type noopHandler struct{}
 
 func (noopHandler) Handle(ctx context.Context, conn *jsonrpc2.Conn, req *jsonrpc2.Request) {}
 
-type readWriteCloser struct {
-	read, write func(p []byte) (n int, err error)
-}
```

Comment:
Instead of modifying `readWriteCloser` so that it would allow us to test if it's been closed, I just used `net.Pipe()` gtom std lib instead.